### PR TITLE
KAFKA-9832: Extend Streams system tests for EOS-beta

### PR DESCRIPTION
**Pull request by mjsax**
Other system tests that need an update.

Call for review @abbccdda @guozhangwang 

System test for both tests passed:
 - https://jenkins.confluent.io/job/system-test-kafka-branch-builder/3890/
 - https://jenkins.confluent.io/job/system-test-kafka-branch-builder/3888/

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTestTest.java [hunk 1]
```diff
@@ -39,6 +41,7 @@ public void verifySmokeTestLogic() {
                                             "nothing:0",
                                             "test",
                                             "test",
+                                            StreamsConfig.AT_LEAST_ONCE,
```
Comment (abbccdda) [inline streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTestTest.java:44]: nit: we could add all 3 modes in this test too

File: streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTest.java [hunk 2]
```diff
@@ -637,13 +637,14 @@ public static Topology getTopology() {
         public static Properties getConfig(final String broker,
                                            final String application,
                                            final String id,
+                                           final String processingGuarantee,
```
Comment (abbccdda) [inline streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTest.java:640]: Reading from the comment, the RelationalSmokeTest has nothing to do with EOS, why do we need to expand this test? cc @vvcephei

Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTest.java:640]: The test had EOS enabled. So I extended it to now run with both EOS versions.

File: streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTestTest.java (hunk 1 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTestTest.java:44]: This is just a "sanity" check using `TopologyTestDriver` -- running it with all 3 modes does not buy us anything as the TTD does not work any different for the 3 modes.

File: streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTest.java (hunk 2 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/tests/RelationalSmokeTest.java:640]: I think it worth adding both eos mode for this test --- if we only have this test with eos-beta there may be some regressions not captured.

In the future, when we deprecate eos-alpha in some release version that requires broker > 2.5 we can remove eos-alpha parameters in all these.

**Review by guozhangwang — COMMENTED** [review]
LGTM overall. Please run system tests with multiple repeats before merging.

**Comment by mjsax** [comment]
Java 8 passed.
Java 11: `org.apache.kafka.connect.mirror.MirrorConnectorsIntegrationTest.testReplication`

Retriggered system test with 20 runs each:
 - https://jenkins.confluent.io/job/system-test-kafka-branch-builder/3891/
 - https://jenkins.confluent.io/job/system-test-kafka-branch-builder/3892/

Retest this please.

**Comment by mjsax** [comment]
Multiple runs of `StreamsSmokeTest ` with `exactly_once` and `exaclty_once_beta` failed -- does not seem to be a kip-447 specific issue. Will investigate.

`StreamsRelationalSmokeTest` passed.

**Comment by mjsax** [comment]
Root cause for the system test issue is an "regression" that we introduced via #8380 -- we set the `transaction.id` incorrectly and this leads to duplicate ID, and thus to `ProducerFencedException`.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java [hunk 3]
```diff
@@ -87,8 +86,6 @@
         this.log = log;
 
         createTaskSensor = ThreadMetrics.createTaskSensor(threadId, streamsMetrics);
-        final String applicationId = config.getString(StreamsConfig.APPLICATION_ID_CONFIG);
-        transactionalId = applicationId + "-" + processId + "-StreamThread-" + threadId.split("-StreamThread-")[1];
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java:91]: @guozhangwang @abbccdda this was the correct way to set the `transaction.id` -- during the refactoring this code was not moved into `StreamsProducer` as it should have

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java [hunk 4]
```diff
@@ -113,7 +113,9 @@ public StreamsProducer(final StreamsConfig config,
                 final String applicationId = config.getString(StreamsConfig.APPLICATION_ID_CONFIG);
                 producerConfigs.put(
                     ProducerConfig.TRANSACTIONAL_ID_CONFIG,
-                    applicationId + "-" + Objects.requireNonNull(processId, "processId cannot be null for exactly-once beta"));
+                    applicationId + "-" +
+                        Objects.requireNonNull(processId, "processId cannot be null for exactly-once beta") +
+                        "-" + threadId.split("-StreamThread-")[1]);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java:118]: Include the thread-index in the `transactional.id` to generate unique IDs for different threads within the same process.

(I stripped out the `"-StreamThread-"` string in contrast to the old code for shorten the ID)

**Comment by mjsax** [comment]
Retriggered 20 runs for `StreamsSmokeTest`: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/3894/

**Comment by mjsax** [comment]
System test build passed.

**Review by abbccdda — APPROVED** [review]

**Comment by guozhangwang** [comment]
LGTM!

**Pull request merged**